packages/core/src/browser/quick-open/quick-title-bar.ts

```diff
@@ -0,0 +1,259 @@
+    }
+
+    get totalSteps(): number | undefined {
+        return this._totalSteps;
+    }
+
+    set buttons(buttons: ReadonlyArray<QuickInputTitleButton> | undefined) {
+        if (buttons === undefined) {
+            this._buttons = [];
+            return;
+        }
+
+        this._buttons = buttons;
+    }
+
+    get buttons() {
+        return this._buttons;
+    }
+
+    private updateInnerTitleText(): void {
+        let innerTitle = '';
+
+        if (this.title) {
+            innerTitle = this.title + ' ';
+        }
+
+        if (this.step && this.totalSteps) {
+            innerTitle += `(${this.step} / ${this.totalSteps})`;
+        } else if (this.step) {
+            innerTitle += this.step;
+        }
+
+        this.titleElement.innerText = innerTitle;
+    }
+
+    // Left buttons are for the buttons dervied from QuickInputButtons
+    private getLeftButtons() {
+        if (this._buttons === undefined || this._buttons.length === 0) {
+            return [];
+        }
+        return this._buttons.filter(btn => btn.side === QuickInputTitleButtonSide.LEFT);
+    }
+
+    private getRightButtons() {
+        if (this._buttons === undefined || this._buttons.length === 0) {
+            return [];
+        }
+        return this._buttons.filter(btn => btn.side === QuickInputTitleButtonSide.RIGHT);
+    }
+
+    private createButtonElement(buttons: ReadonlyArray<QuickInputTitleButton>) {
+        const buttonDiv = document.createElement('div');
+        buttonDiv.style.display = 'inline-flex';
+        for (const btn of buttons) {
+            const aElement = document.createElement('a');
+            aElement.style.width = '16px';
+            aElement.style.height = '16px';
+            aElement.tabIndex = 0;
+            if (btn.iconClass) {
+                aElement.classList.add(...btn.iconClass.split(' '));
+            }
+
+            if (btn.icon !== '') {
+                aElement.style.backgroundImage = `url(\'${btn.icon}\')`;
+            }
+
+            aElement.classList.add('icon');
+            aElement.style.display = 'flex';
+            aElement.style.justifyContent = 'center';
+            aElement.style.alignItems = 'center';
+            aElement.style.cursor = 'pointer';
+            aElement.tabIndex = this.tabIndex;
+            aElement.title = btn.tooltip ? btn.tooltip : '';
+            aElement.onclick = () => {
+                this.onDidTriggerButtonEmitter.fire(btn);
+            };
+            aElement.onkeyup = event => {
+                if (event.code === 'Enter') {
+                    aElement.click();
+                }
+            };
+            buttonDiv.appendChild(aElement);
+            this.tabIndex += 1;
+        }
+        return buttonDiv;
+    }
+
+    private createTitleBarDiv() {
+        const div = document.createElement('div');
+        div.style.display = 'flex';
+        div.style.flexDirection = 'row';
+        div.style.fontSize = '13px';
```

Comment:
please use defined theia css variables for font size and padding, try `--theia-ui-font-size0` and `--theia-ui-padding`

- font size and padding of toolbar looks odd:
  - font size should be smaller
  - top and bottom padding should be bigger, there is almost no-spare right now
- buttons should be aligned with the text, right now it looks like buttons aligned to the top, but text is centered

Reply:
done